resource/graph_node.go

```diff
@@ -189,6 +193,32 @@ func (w *GraphNode) NeedsReconfigure() bool {
 	return !w.markedForRemoval && w.needsReconfigure
 }
 
+func (w *GraphNode) timesReconfiguredErr() error {
+	var noun string
+	if w.IsUninitialized() {
+		noun = "configuration"
+	} else {
+		noun = "reconfiguration"
+	}
+	return errors.Errorf(
+		"%s error: reached max of %d %s attempts for %s",
+		noun,
+		MaxReconfigAttempts,
+		noun,
+		w.config.ResourceName(),
+	)
+}
+
+// ValidateReconfigure returns whether or not the resource is able to be reconfigured
+// based on how many previous attempts were made— nil if we are able to reconfigure,
+// or the appropriate error with the reason why we cannot reconfigure.
+func (w *GraphNode) ValidateReconfigure() error {
```

Comment:
[nit] Can we call this method and the similarly named function in the resource manager `C(c)heckReconfigureAttempts` instead? Validation is a separate concept within the resource graph/manager and I sense I'll be confused when I look at the names of these functions in two months.